pymeasure/instruments/agilent/agilentL4534A.py

```diff
@@ -0,0 +1,425 @@
+        """,
+        validator=sample_count_function,
+        values=SAMPLE_COUNT_VALUES
+    )
+
+    pre_trigger_samples = Instrument.control(
+        "CONF:ACQ:SPR?",
+        "CONF:ACQ:SPR %d",
+        """
+        Control number of samples that will captured pre-trigger
+        (int from 0 to samples_per_record-4 in multiples of 4).
+        """,
+        validator=sample_count_function,
+        values=PRE_TRIG_VALUES
+    )
+
+    number_of_records = Instrument.control(
+        "CONF:ACQ:REC?",
+        "CONF:ACQ:REQ %d",
+        """
+        Control number of of records that will be captured before arming is disabled
+        (int from 1 to 1024).
+        """,
+        validator=strict_range,
+        values=RECORD_COUNT_VALUES
+    )
+
+    trigger_holdoff = Instrument.control(
+        "CONF:ACQ:THOL?",
+        "CONF:ACQ:THOL %e",
+        """Control minimum time between triggers, during which trigger inputs are ignored,
+          in seconds (float from 0 to 10).
+        """,
+        validator=strict_range,
+        values=TRIGGER_HOLDOFF_VALUES
+    )
+
+    trigger_delay = Instrument.control(
+        "CONF:ACQ:TDEL?",
+        "CONF:ACQ:TDEL %e",
+        """
+        Control delay time after trigger before acquistion starts,
+        in seconds (float from 0 to 3600).
+        """,
+        validator=strict_range,
+        values=TRIGGER_DELAY_VALUES
+    )
+
+    maximum_samples = Instrument.measurement(
+        "CONF:ACQ:SCO:MAX?",
+        """Get Maximum number of samples that can be used in a record."""
+    )
+
+    acquisition = Instrument.control(
+        "CONF:ACQ:ATTR?",
+        "CONF:ACQ:ATTR %s",
+        """
+        Control Acquistion settings
+
+        Represented as dictionary containing:
+        - sample_rate
+        - samples_per_record
+        - pre_trig_samples_per_record
+        - num_records
+        - trigger_holdoff
+        - trigger_delay
+        """,
+        set_process=_set_acq_config_process,
+        get_process=_get_acq_config_process,
+        validator=_validate_acq_config
+    )
+
+    def arm(self) -> None:
+        """
+        Trigger ARM condition in software
+        """
+        self.write('ARM')
+
+    def init(self) -> None:
+        """
+        Initialize measurement with current configuration.
+
+        note::
+        This puts the instrument in acquisition state and waits for arm condition.
+        - If arm is immediate, it will wait for trigger.
+        - If both arm and trigger are immediate, it will immediately start capturing data.
+
+        Instrument will singal complete once capture is finished.
+        """
+        self.write('INIT')
+
+    def auto_zero(self, channels=[1, 2, 3, 4]):
+        """
+        Auto-zero the inputs, temporarily loading new offsets for the current config.
+
+        :param channels: list of channels to auto-zero
+        :return: 0 if auto-zero completed successfully, Otherwise, an error ocurred
```

Comment:
In python, it is better to raise an error, if one happened.
Some idea:
~~~python
If returned_value != 0:
    Raise SomeError("AutoZero failed.")
~~~